### PR TITLE
Token Revocation Feature

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
This PR will include following.

- Installing org.wso2.carbon.event.output.adapter.server.feature.group and org.wso2.carbon.event.output.adapter.server.feature.group features in p2 profiles for key-manager profile.
- Adding carbon.identity-data-publisher-oauth.version to master pom file.
- Adding token revocation default topic name in jndi.properties file.
- Integration Test class for this feature.
- Resource config files which are relevant for the Integration test class.
- etcdMock war which will be deployed when the test is running.
- Adding the test to testing.xml file.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/revokejwttest/jndi.properties [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/revokejwttest/jndi.properties]: Let's update the licence year

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+        try {
+            //create a subscriber to jwRevocation topic in the JMS broker
+            createJMSConnection();
+        } catch (NamingException e) {
+            log.error("Error while Creating JMS Connection - Naming Exception:", e);
+        } catch (JMSException e) {
+            log.error("Error while Creating JMS Connection - JMS Exception:", e);
+        }
+
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: Remove the additional lines

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        //Checking the message content of the received JMS message
+        Message message = consumer.receive();
+        if (message instanceof MapMessage) {
+            MapMessage mapMessage = (MapMessage) message;
+            assertEquals(mapMessage.getString("revokedToken"), jtiExtracted);
+        }
+        topicConsumerSession.close();
+        topicConnection.close();
+
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: Same here

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        log.info("checkJMSTopicForRevokedJTITTestCase  Finished");
+    }
+
+    @AfterClass(alwaysRun = true)
+    void destroy() throws Exception {
+        super.cleanUp();
+    }
+
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: Same here

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+        jti= "2f3c1e3a-fe4c-4cd4-b049-156e3c63fc5d";
+        assertEquals(retrievedETCDJTIKey, jti);
+        log.info("checkETCDForRevokedJTITTestCase  Finished");
+    }
+
+    @Test(groups = { "wso2.am" }, description = "JWT revocation test", dependsOnMethods = "revokeRequestTestCase")
+    public void checkJMSTopicForRevokedJTITestCase() throws Exception {
+        log.info("checkJMSTopicForRevokedJTITTestCase  Initiated");
+
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: Lets remove the extra lines in all the classes

File: modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/revokejwttest/jndi.properties (hunk 1 above)
Comment (CrowleyRajapakse) [reply modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/revokejwttest/jndi.properties]: resolved

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java (hunk 3 above)
Comment (CrowleyRajapakse) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: resolved

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java (hunk 4 above)
Comment (CrowleyRajapakse) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: resolved

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java (hunk 5 above)
Comment (CrowleyRajapakse) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: resolved

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java (hunk 2 above)
Comment (CrowleyRajapakse) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: resolved

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+    @Test(groups = { "wso2.am" }, description = "JWT revocation test", dependsOnMethods = "revokeRequestTestCase")
+    public void checkETCDForRevokedJTITestCase() throws Exception {
+        log.info("checkETCDForRevokedJTITTestCase  Initiated");
+        String etcdEndpointURLs = "https://localhost:9943/etcdmock/v2/keys/jti/2f3c1e3a-fe4c-4cd4-b049-156e3c63fc5d";
+        URL etcdEndpointURL = new URL("https://localhost:9943/etcdmock/v2/keys/jti/2f3c1e3a-fe4c-4cd4-b049-156e3c63fc5d");
+        Map<String, String> authenticationRequestHeaders = new HashMap<>();
+        authenticationRequestHeaders.put("Content-Type", "application/x-www-form-urlencoded");
+        String retrievedETCDJTIKey = "";
+        String input ="";
```
Comment (isharac) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: need a space after =

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+        try {
+            HTTPSClientUtils.doPost(etcdEndpointURL, input, authenticationRequestHeaders);
+            JSONObject etcdResponse = new JSONObject(
```
Comment (isharac) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: need a space before {

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+                        if (childKey.equalsIgnoreCase("key")) {
+                            retrievedETCDJTIKey = childObject.getString(childKey).substring(5);
+                        }
+                    }
+                }
+            }
+        } catch (JSONException e) {
+            Assert.fail("Should not throw any exceptions" + e);
+        }
+        jti= "2f3c1e3a-fe4c-4cd4-b049-156e3c63fc5d";
```
Comment (Meruja) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: Need a space before =

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java (hunk 6 above)
Comment (CrowleyRajapakse) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: resolved with https://github.com/wso2/product-apim/pull/7041

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java (hunk 8 above)
Comment (CrowleyRajapakse) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTRevocationTestCase.java]: resolved with https://github.com/wso2/product-apim/pull/7041

**Pull request merged**